### ISSUE TITLE
Autoclosing issues/PRs labeled "stalled" after 30 days

### ISSUE THREAD

**Issue by jasnell** [nodejs/github-bot#261]
Would it be possible for the bot to automatically close issues/prs 30 days after they have manually been labeled "stalled", along with a comment added as to why it is being closed. I don't want to get into auto-labeling stalled issues just yet but automatically cleaning up those that have been labeled would be good.

**Comment by phillipj** [comment]
Cool idea. I would assume it's possible for us to resolve PRs by that label once a day or similar.

Has there been any discussion in core to use GitHub Actions for things like this? I stumbled upon the [stale](https://github.com/actions/stale) actions recently. Taking some inspiration from that to close PRs with a specific label might not be too far fetched 🤷‍♂️

**Comment by mmarchini** [comment]
We use an action for that on nodejs/build: https://github.com/nodejs/build/blob/master/.github/workflows/stale.yml. I don't think implementing this in the bot is worth it.

**Comment by phillipj** [comment]
That stale action' [`only-labels` option](https://github.com/actions/stale/blob/master/action.yml#L30) seems to be what we're looking for?

```yaml
only-labels:
  description: 'Only issues or pull requests with all of these labels are checked if stale. Defaults to `[]` (disabled) and can be a comma-separated list of labels.'
```

**Comment by phillipj** [comment]
@jasnell would it be valuable if I confirmed the [stale GitHub Action](https://github.com/actions/stale) indeed solves what you're asking? Or are we shying away from Actions use in core for some reason?

**Comment by jasnell** [comment]
@phillipj ... That would be great. The one caveat is that we really ought to avoid automatic stale marking.

**Comment by phillipj** [comment]
Noted. I'll setup a test repo to see how it behaves and report back.

Thx for the quick reply!

**Comment by mmarchini** [comment]
@phillipj you can also use nodejs/node-auto-test if you want

**Comment by phillipj** [comment]
**TLDR;** it works.

Had to fork the stale Action and make minor modifications, primarily because we want to perform the `stalled` labelling ourselfs rather automatic via the Action.

I agree with @mmarchini on his point using an Action for this vs our beloved github-bot. Feels a lot simpler to grasp due to the reduced code base, more focused, and likely easier for core collaborators to tweak or throw away when not wanted anymore.

Other than doing a few example runs in [nodejs/node-auto-test](https://github.com/nodejs/node-auto-test), does the idea of using an action for this sound reasonable?

Refs: [stale Action modifications](https://github.com/phillipj/close-stale/commit/f64cddbdc2c5bc09b58d0f58d129417d9e614233), [Action config](https://github.com/phillipj/stale-actions-without-label-config/blob/master/.github/workflows/stale.yml), [closed issue](https://github.com/phillipj/stale-actions-without-label-config/issues/4) and [not closed issue](https://github.com/phillipj/stale-actions-without-label-config/issues/1).

**Comment by mmarchini** [comment]
imo if you can show an issue getting closed and one not getting closed/labeled in any repository  (and you already did that) that's good enough to send a PR to nodejs/node. Also, I don't see a problem with forked Actions, they were built so they could be forked.

**Comment by phillipj** [comment]
Cool!

I'll see if I can find different set of changes that could land into the upstream stale Action project, as opposed to forking in and deleting a few lines we don't want to run.

Will open a PR against nodejs/node afterwards.

**Comment by mmarchini** [comment]
Worst case we could use a really big number (like 9999999, which is more than two thousand years) of days to mark an issue as stale :shrug:

**Comment by phillipj** [comment]
_Latest update on this;_ changes has landed in the upstream stale action project. That means we can use that GitHub Action out of the box.

Awaiting them to build & push a new version, before opening a PR to nodejs/node.

**Comment by phillipj** [comment]
Closing this as it feels like we've settled for realising this with a GitHub Action in nodejs/node instead of our beloved @nodejs-github-bot.
